scalar/interval.go

```diff
@@ -0,0 +1,254 @@
+package scalar
+
+import (
+	"encoding/json"
+
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type MonthInterval struct {
+	Int
```

Comment:
IMO it requires a separate `String` method

Reply:
Done, since it's passed in from Arrow side as `monthIntervalData` struct I also used it in the stringer.